### DISCUSSION TITLE
FEPX Fasets issue

### DISCUSSION THREAD

**Discussion by BK-456** [neperfepx/FEPX#105] [Q&A] [answered]
Hello,

Currently, I am experiencing the following error while trying run a simulation:

==========================     F   E   P   X   ==========================
Info   : A finite element software package for  polycrystal plasticity.
Info   : Version 1.3.0
Info   : Running on 1 cores.
Info   : <https://fepx.info>
Info   : Copyright (C) 1996-2021, DPLab, ACME Lab.
Info   : ---------------------------------------------------------------
Info   : Start time: 2023-8-25 at 14:18
Info   : Loading simulation...
Info   :   [i] Parsing file `simulation.config'...
Info   :   - Material parameters:
Info   :     > Number of phases: 1
Info   :     > phase 1 - crystal type:   FCC
Info   :     > m:               0.500000E+00
Info   :     > gammadot_0:      0.100000E-02
Info   :     > h_0:             0.140000E+03
Info   :     > g_0:             0.500000E+02
Info   :     > g_s0:            0.800000E+02
Info   :     > n:               0.140000E+01
Info   :     > c11:             0.750000E+05
Info   :     > c12:             0.310000E+05
Info   :     > c44:             0.200000E+05
Info   :   - Boundary conditions:
Info   :     > Uniaxial grip
Info   :     > Strain targeting, constant strain rate
Info   :     > Strain rate:   0.500000E-02
Info   :     > Loading direction: z
Info   :     > Loading face: z1
Info   :   - Deformation history:
Info   :     > Number of strain steps: 8
Info   :   [i] Parsed file `simulation.config'.
Info   :   [i] Parsing file `simulation.msh'...
Info   :   - Mesh parameters:
Info   :     > Node number: 15474
Info   :     > Elt  number: 9363
Error  :     > 6 surfaces necessary in $Fasets.
========================================================================

I am attaching the visualization of the tessellation file here for your reference. Kindly help me to resolve the issue.
![columnar_Poly_refined_image3](https://github.com/neperfepx/FEPX/assets/135224657/c8baac7b-1e2b-469f-a3a1-95a09475ebe5)

![Error](https://github.com/neperfepx/FEPX/assets/135224657/f9bcd4f8-1071-4a3a-b5bd-029d730066ad)

**Comment by kight55** [comment]
The simulated RVE must have six faces, as you have dug a hole in the middle, it is not six faces anymore. From this post on page 7, Simulating 2d microstructure having notch geometry, you can learn more

**Reply by mkasemer** [reply]
This is something that will be addressed in upcoming versions of FEPX. For now, you can try deleting some of the potentially-unnecessary faces in the mesh file as a ``hack''. Be sure to keep the domain surfaces, especially those on which boundary conditions are applied.

**Reply by BK-456** [reply]
Thank you for your response. 
How do I "hack" away the potentially unnecessary fasces? The displayed domain has 7 surfaces, so 7 are listed in the mesh file. Consider that I wish to replace "cut1" with "y0" or a subsection"y0_1" inoder to make it 6 surfaces. Simply reducing the fasets from "7" to "6" and replacing the "cut1" section with "y0" does not fix the problem.
Please indicate how to modify the mesh file as a "hack" using an example. I have attached the mesh file in text format for your review.
[simulation.msh.txt](https://github.com/neperfepx/FEPX/files/12515868/simulation.msh.txt)

**Reply by kight55** [reply, marked as answer]
You can try changing only the data under fasets, changing fasets from "7" to "6", and deleting all data in cut1 under fasets,

**Reply by BK-456** [reply]
Thank you, that was exactly what I needed.
